### PR TITLE
.htaccess rewrite rule, dev flag, README.md, toolbar.yml

### PR DESCRIPTION
**Pull request by SpiritLevel**
I think you meant to put ".org" instead of ".com" on the RewriteCond.

Also, I don't think you need the "s?" in the RewriteCond: you only want to redirect doc.silverstripe.org to docs.silverstripe.org, and just let docs.silverstripe.org pass through ?

### CODE REVIEW

**Comment by SpiritLevel** [comment]
I updated README.md to reflect the current state of things. I intend to look at putting a flag in `RefreshMarkdownTask` to prevent the deletion of the git repos. Also, there is a bit of redundancy in the YAML files (ie: docsviewer.yml and docs-repositories.yml) which should be eliminated if possible. Lastly, I think the two `sake` commands should be combined into a single command, and perhaps `RebuildLuceneDocsIndex` could be renamed at the same time to something like `RebuildDocsIndex`; whether or not Lucene is used is purely an internal implementation issue and should not explicitly be in the name of the task. :)  Just my opinions!

**Comment by camfindlay** [comment]
1) I meant to put .com
2) RebuildLuceneDocsIndex is part of docsviewer and should be it's own task

**Comment by SpiritLevel** [comment]
@camfindlay 

1) Thought so ;)

2) Yup. But, for installing docs, it would be nice to have a single `sake RefreshDocs flush=1` command that runs `RefreshMardownTask` and `RebuildDocsIndex`. Just a thought.

File: .htaccess [hunk 1]
```diff
@@ -29,8 +29,8 @@ ErrorDocument 500 /assets/error-500.html
 <IfModule mod_rewrite.c>
 	RewriteEngine On
 
-	RewriteCond %{HTTP_HOST} ^docs?.silverstripe.com$ [NC]
-    RewriteRule ^(.*)$ https://docs.silverstripe.org/$1 [L,R=301]
+	RewriteCond %{HTTP_HOST} ^doc.silverstripe.org$ [NC]
```
Comment (camfindlay) [inline .htaccess]: Leave this as is please :D

**Comment by camfindlay** [comment]
Happy to leave the scripts as two separate ones for now. More pressing matters to address.

**Comment by SpiritLevel** [comment]
OK...just add dev flag and set .htaccess RewriteCond to include the "s?".

**Comment by SpiritLevel** [comment]
Oh, I noticed an issue when running RefreshMarkdownTask in a fresh clone & composer install of doc.silverstripe.org. The assets/src/framework_* folders are not initially present and so DocumentationManifest complains about their absence. So, some kind of manifest refresh needs to be done but I am not sure of the best way to do this...being a newbie!

**Comment by SpiritLevel** [comment]
I'll update README.md

**Comment by camfindlay** [comment]
updating the manifest is just calling a flush

**Comment by SpiritLevel** [comment]
OK...README.md updated as well as toolbar.yml :)

**Comment by SpiritLevel** [comment]
Do you mean just including the `flush=1` flag ? I'm a bit confused since this didn't work and I can't see anywhere in `RefreshMarkdownTask` where the manifest is updated. Will try again and look a it bit more...

**Comment by SpiritLevel** [comment]
OK. `flush=1` definitely doesn't do what we need.

**Comment by camfindlay** [comment]
Sorry misunderstood what you mean. This is expected behaviour of docsviewer when you're told it there should be markdown source and there is none. YOu need to run the build task , get the md source and that will sort of that issue. It could be handled more elegantly in docsviewer when it cannot find a referenced source folder

File: app/code/RefreshMarkdownTask.php [hunk 2]
```diff
@@ -106,6 +106,10 @@ private function cloneRepository(array $repository)
      */
     private function cleanRepository(array $repository)
     {
+	
+	// if the dev=1 flag is used when invoking RefreshMarkdownTask the git repos are kept.
+	if( isset($_REQUEST['dev']) && $_REQUEST['dev'] ) return;
```
Comment (camfindlay) [inline app/code/RefreshMarkdownTask.php]: Probably needs to change the git clone command too to ensure you get the full history not just the shallow clone.

File: app/_config/toolbar.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ Only:
   environment: dev
 ---
 GlobalNav:
-  hostname: '//ssorg.dev/'
+  hostname: '//www.silverstripe.org/'
```
Comment (camfindlay) [inline app/_config/toolbar.yml]: thanks

File: .htaccess [hunk 4]
```diff
@@ -29,8 +29,8 @@ ErrorDocument 500 /assets/error-500.html
 <IfModule mod_rewrite.c>
 	RewriteEngine On
 
-	RewriteCond %{HTTP_HOST} ^docs?.silverstripe.com$ [NC]
-    RewriteRule ^(.*)$ https://docs.silverstripe.org/$1 [L,R=301]
+	RewriteCond %{HTTP_HOST} ^docs?.silverstripe.org$ [NC]
```
Comment (camfindlay) [inline .htaccess]: Please change this back

**Comment by SpiritLevel** [comment]
But, the error occurs while running the build task `RefreshMarkdownTask` so running it doesn't fix the problem. I had to manually add the missing directories to get it to run. Perhaps we should include in `RefreshMarkdownTask` something to create the needed directories if not present or remove the code that checks for their presence.

File: app/code/RefreshMarkdownTask.php (hunk 2 above)
Comment (SpiritLevel) [inline app/code/RefreshMarkdownTask.php]: Darn...there goes some of the efficiency gain you made :(

File: .htaccess (hunk 4 above)
Comment (SpiritLevel) [inline .htaccess]: Sorry...not quite understanding....what do you want it changed back to ?

Comment (camfindlay) [inline .htaccess]: the condition is if request comes in on our .com domain eg docs.silverstripe.com it redirects to the correct docs.silverstripe.org domain. What you have here will create an infinite loop when you get it on the actual production server.

Comment (SpiritLevel) [inline .htaccess]: Ahh...I see what you are are wanting .....your right about the infinite redirects....what I had was ok when the "s?" was removed....but now it loops...oopsie.

So, you want the .com and the "s?" back in ?

Comment (camfindlay) [inline .htaccess]: Put it as it was when this code was merged in to master ;)

File: app/code/RefreshMarkdownTask.php (hunk 2 above)
Comment (SpiritLevel) [inline app/code/RefreshMarkdownTask.php]: Done.

**Comment by SpiritLevel** [comment]
@camfindlay OK, when `RefreshMarkdownTask` is run, it is run within `doc.silverstripe.org` and hence `docsviewer` first gets fired up, which immediately reads `docsviewer.yml`. Then, upon constructing a `DocumentationManifest`, the constructor checks to see if the files registered in `docsviewer.yml` are actually present. If they aren't, an error is thrown, which is what happens running the task with a fresh clone and composer install of  `doc.silverstripe.org`. This occurs even before `docs-repositories.yml` is read.

This cannot be fixed easily by simply creating the needed directories in `RefreshMarkdownTask` since `DocumentationManifest` is run first. 

The easiest fix for this directory problem is to just create empty directories `assets/src/framework_master` etc. in the github repo `doc.silverstripe.org` so that they exist after the initial cloning. I'll implement this fix now but there must be a better solution.

Even with the above solution to the directory problem, the uncomfortable duplication of config information would remain. I'm not sure of the best way to handle this but it definitely needs fixing: one only wants a **single** place where the documentation files and versions are registered. This seems to suggest additional reorganisation of `docsviewer` is needed....

Ideas/suggestions anyone ? @dhensby ?

**Comment by SpiritLevel** [comment]
@camfindlay I noticed the site has reverted. I am on standby to do more once sorted :)

**Comment by camfindlay** [comment]
@SpiritLevel and... we're back up on the new servers. If you want to continue. There is also some noise over on the docsviewer module you might want to check in on.

**Comment by SpiritLevel** [comment]
@camfindlay Ok. Where do you want to go from here ? Should I separated these into separate PRs ? See also https://github.com/silverstripe/silverstripe-cms/issues/1407

**Comment by camfindlay** [comment]
Separate PRs would be useful and easier to test

**Comment by SpiritLevel** [comment]
OK, will separate...

**Pull request closed without merging**